---
title: "Orthology inference is unable to recover fast-evolving, metazoan single-copy orthologs (and why this is good)"
author: ""
date: ""
output: 
  html_document:
    code_folding: hide
    code_download: true
    toc: true                  # table of content true
    toc_depth: 3               # upto three depths of headings (specified by #, ## and ###)
    toc_float: true
    number_sections: false     # if you want number sections at each table header
    theme: yeti              # many options for theme, this one is my favorite.
    highlight: tango           # specifies the syntax highlighting style
    #css: my.css               # you can add your custom css, should be in same folder
    # override defaults by calling with custom params from command line https://bookdown.org/yihui/rmarkdown/params-knit.html 15.3.2
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Introduction

Two genes are known as orthologous if they originated during an event of speciation, where a species splitting in two 
resulted in every single gene diverging in two copies. Instead, genes that their ancestry is based on a duplication event are called paralogous genes. 

Identifying orthologous genes is an important step in many biological analyses, such as genome annotation, comparative genomics and phylogenomics. Specifically in phylogenomics, orthologous genes that are found in single copies (no duplication after speciation) across multiple species are necessary, because their evolutionary history can provide insights about speciation events and thus, evolutionary history of the species themselves.

Inferring orthology relationships among genes is not a trivial task. Various orthology inference methods have been developed, mostly involving sequence similarity and clustering. Furthermore, the accuracy of an orthology analysis is hard to assess because no ground truth can be known about the evolutionary history of genes.

It is common practice in evolutionary genomics to employ simulation experiments in order to verify and benchmark newly developed methods. In a simulation framework, sequences can evolve under a user-specified model where every parameter is known and controlled, and all required information can be recorded. Even if models are oversimplifying the true evolutionary events, simulations are often a valuable tool to understand the behaviour of new models and algorithms.

## Scope of this work

Here, simulation experiments are employed to assess the ability of OrthoFinder to recover single-copy genes across metazoan species. Artificial Life Framework (ALF, [ref]) is used to simulate genome evolution from 1 ancestral to 44 target genomes under a given guide tree. The 44 target genomes correspond to 44 real metazoan genomes and the guide tree represents the  phylogenetic relationships of these species.

Multiple simulation experiments were performed under different evolutionary scenarios. Before each repeat, all branch lengths of the guide tree was multiplied by a scalar that increased or decreased the total tree length. This means that genomes and their genes will have less or more time to evolve, and thus orthologue genes will be more or less diverged from each other.

In each repeat, the starting genome comprised of 10,000 genes that evolve throughout the guide tree under the LG model exchangeabilities. Zero gene duplication and gene loss rates were specified, so each of the 44 resulting proteomes will also contain 10,000 single-copy genes. 
Every gene from one proteome will have exactly one orthologue in each of the rest 43 proteomes, for a total of 10,000 single-copy groups of orthologues with 44 genes each.

Orthofinder was applied in the simulated proteomes and the expected result was to recover 10,000 single-copy orthogroups with 44 genes each. Any deviation from this will be due to OrthoFinder's inability to correctly recover a fraction of these groups. In the end, the goal of this work is to assess OrthoFinder's performance in recovering metazoan single-copy orthologues in relation to how divergent these are from each other. This is of great interest because there are known instances of Metazoa species that evolve in faster rates than others, so their orthologue genes will be more diverged compared to slower-evolving species. Nevertheless, correctly identifying orthology relationships is critically important in the context of reconstructing the animal Tree of Life.

## Simulating Genome Evolution

Evolution from one ancestral to 44 terminal proteomes under a given guide tree was simulated using ALF. Different simulation experiments were performed under 200 different evolutionary scenarios, each time resulting in 44 proteomes with 10,000 genes each. In each of the 200 repeats, the guide tree was down- or up-scaled to allow for less or more evolutionary time, and also a different parameter for rate variation among sites (alpha) was given. 

OrthoFinder was applied to each of the 200 44-proteome simulated datasets and various result metrics were recorded, such as the total number of resulting orthogroups, the number of genes not assigned to any orthogroup, the mean orthogroups size etc. The results are summarised in the following dataframe:

```{r}
ortho_results <- data.table::fread("data/orthosims_stats.csv")
head(ortho_results)
```

There are 9 columns in this matrix:

- **SimID** the unique ID of each simulation experiment
- **alpha** the alpha parameter for rate variation among sites used in each simulation
- **scalar** the number with which all branch lengths of the guide were multiplied to increase/decrease evolution time during the simulation
- **orthogroups** how many orthogroups were retained by OrthoFinder from the simulated proteomes (expected 10,000)
- **all_species_single_copy** how many of the recovered orthogroups contained exactly one gene from all 44 species (expected 10,000)
- **unassigned_genes** how many of the 440,000 input genes (44 * 10,000) were not assigned by OrthoFinder to any of the resulting orthogroups (expected 0)
- **percentage_unassigned** proportion of the unassigned genes (unassigned_genes / 440,000) 
- **mean_orthogroup_size** the average number of genes contained in one orthogroup (expected 44)
- **median_orthogroup_size** half of the orthogroups have this number of genes or more, and half of the orthogroups have this number of genes or fewer (expected 44)

<br>

In the controlled environment of simulations, various aspects of the evolutionary process that are not known in real world can be recorded. This knowledge provides a basis on which the accuracy of methods, algorithms and software can be assessed. In our example, the "true" result of the evolution is 10,000 single-copy groups of orthologues, contaning 44 genes each. We can assess OrthoFinder's accuracy under different simulation settings by comparing the resulting orthogroups to this expected result. 

<br>

Let's see first how much the number of estimated orthogroups deviated from the expected 10,000 under different combinations of guide tree length and alpha parameter for rate variation among sites:

<br>

```

```

<br>

This plot can be divided in three regions, in respect to the scalar that guide tree length is multiplied before simulation. For scalars <2 the number of inferred orhtogroups does not deviate much from 10,000 regardless of the alpha value. For scalars between 2-5 there is much variance between results for the same scalar, depending on the value of the alpha paramaters. Larger alphas lead to greater number of orthogroups for the same scalar. However, the estimated number of orthogroups never goes beyond 85,000. For scalars >5 the results become mixed. As it can be seen in the bottom-right part of the scatterplot, large scalars and large alphas can drop the estimated number of orthogroups closer to 10,000. What does this mean? Let's see the number of unassigned genes plot to help us understand what is going on:

<br>

```

```

<br>
<br>

Here the trend becomes clearer. For shorter guide trees almost all genes are assigned to some orthogroup but as the guide tree become larger, the number of unassigned genes is analogous to the alpha parameter. For the same guide tree length, larger alpha leads to more genes not placed in any orthogroup, probably because they evolved faster than their orthologues and became so diverged that their relationship could not be recognised by OrthoFinder anymore. We can see now that for large guide trees and large alphas (top-right corner), almost 400,000 of the 440,000 genes were left without an orthogroup. So, the small number of orthogroups that was observed in the previous plot for this setting was seen because the majority of the input genes were too distant from each other to be placed in an orthogroup.

<br>

Another interesting thing to look at is how many of the recovered orthogroups are all-species single-copy:

<br>

```

```

<br>

Here the expected value is 10,000 since this is the number of genes in the starting proteome of the simulated evolution, and no gene duplications or losses are allowed. We see that only when these initial genes evolve for very little time we are able to estimate a total number of single-copy groups that approximates the desired value. As guide trees grow larger, these estimates quickly become dependent on the alpha paramter of rate variation among sites. For larger alphas, we lose ability to recover even 25% of the desired amount of single-copy orthogroups when the scaler becomes >2.5. For smaller alphas, we are able to recover more than half of the 10,000 single-copy orthogroups for scalars <5.

<br>
<br>

Now let's look at the last plot, which shows how mean orthogroup size varies among simulations:

<br>

```

```

<br>

Again the trend is clear. Here, the expected value is 44, equal to the number of species used in the simulations. In smaller scalars, the mean size of each orthogroup is very close to 44. As guide trees become longer, the mean orthogroup size depends highly on the value of the alpha parameter, until we get to scalars >8 where mean sizes are <5 no matter the value of alpha.

<br>
<br>

## Verdict and next steps

It is clear that when genes become divergent, either because they evolve fast or because they evolve for more time, OrthoFinder becomes less and less able to correctly recover our 10,000 44-species single-copy orthogroups. Even though the data used come from simulations which oversimplifies true events, the remarks will most likely hold for real data as well. Events such as gene duplication, gene loss, horizontal gene transfer, incomplete lineage sorting etc. complicate the evolutionary process and make the accurate inference of single-copy orthogroups much more than in the "ideal" simulation environment. 

<br>

More things that would be interesting to look at in the results of OrthoFinder in simulated proteomes are:
<br>


- How many orthogroups are recovered  per species and how many genes from each species stay unassigned. It would be interesting to see how OrthoFinder places genes from faster-evolving species into orthogroups compared to slower-evolving ones. Since the 44 species are directly linked to real metazoan species, we can see in which groups of animals OrthoFinder becomes more sensitive.

- Apart from looking just at the mean orthogroup size of each run, it would be interesting to look at the whole distribution of orthogroup sizes, and how lumped together or fragmented they are under different simulation settings. This pattern can be directly linked to gene content phylogeny, since as orthogroups start to get more fragmented, this split might be phylogenetically informative in the context of gene presence/absence.
